plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Vesting.hs

```diff
@@ -136,13 +144,20 @@ validatorScriptHash =
     . Ledger.scriptAddress
     . validatorScript
 
+{-# INLINABLE mkValidator #-}
 mkValidator :: Vesting -> VestingData -> () -> PendingTx -> Bool
-mkValidator d@Vesting{..} VestingData{..} () p@PendingTx{pendingTxValidRange = range} =
+mkValidator d@Vesting{..} VestingData{..} () ptx@PendingTx{pendingTxValidRange = range} =
     let
+        payBack :: Value
+        payBack = mconcat (map pendingTxOutValue (getContinuingOutputs ptx))
```

Comment:
I think we could do with a version of `valuePaidTo` that takes a script address. We could equally do with `valuePaidFrom`. Then `amountSpent` becomes 
```valuePaidTo scriptAddr `minus` valuePaidFrom scriptAddr```
which I think is nice and clear.

Reply:
Do you think those functions should be added to the library, or just for this contract?

Reply:
Yes, to the library!

Reply:
Oh, I don't think we did this in the end. I still think this is a good idea.